pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -101,6 +115,11 @@ func (cd *clusterDeploy) doSync(cluster *v3.Cluster) error {
 	if err != nil {
 		return err
 	}
+
+	if cluster.Status.AgentImage != "" && !agentImagesCached(cluster.Name) {
+		cd.cacheAgentImages(cluster.Name)
```

Comment:
we need to handle error returned from cacheAgentImages

Reply:
ah that was unintentional. sorry will fix